main/src/ca/uwaterloo/flix/api/lsp/Indexer.scala

```diff
@@ -415,14 +415,15 @@ object Indexer {
     * Returns a reverse index for the given type `tpe0`.
     */
   private def visitType(tpe0: Type): Index = tpe0 match {
-    case Type.Var(_, _, _, _) => Index.empty
+    case _: Type.MaybeKindedVar => Index.empty
     case Type.Cst(tc, loc) => tc match {
       case TypeConstructor.RecordExtend(field) => Index.occurrenceOf(tc, loc) ++ Index.useOf(field)
       case TypeConstructor.SchemaExtend(pred) => Index.occurrenceOf(tc, loc) ++ Index.useOf(pred)
       case _ => Index.occurrenceOf(tc, loc)
     }
     case Type.Lambda(_, tpe) => visitType(tpe)
     case Type.Apply(tpe1, tpe2) => visitType(tpe1) ++ visitType(tpe2)
+    case Type.Ascribe(tpe, _) => visitType(tpe)
```

Comment:
In the indexer, it might be better to throw exceptions for the Ascribe and UnkindedVar cases?